### PR TITLE
Extract column e2e

### PR DESCRIPTION
**Pull request by ranquild**
Epic https://github.com/metabase/metabase/pull/39162
Testing plan https://github.com/metabase/metabase/issues/39291

Checks:
- Each option type (unit)
- Breakout and aggregation columns
- Editing the generated expression in the notebook editor
- Duplicate column names
- Using localized values for temporal units

Doesn't check:
- Column is appended to the right - that's not the case now
- Different viz settings for each option - that's not the case now and I'm not sure if everything from the testing plan is correct about this

### CODE REVIEW

File: e2e/test/scenarios/visualizations-tabular/drillthroughs/column_extract_drill.cy.spec.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    cy.intercept("POST", "/api/dataset").as("dataset");
+    cy.intercept("POST", "/api/card/*/query").as("cardQuery");
+  });
+
+  describe("date columns", () => {
+    describe("should add a date expression for each option", () => {
+      DATE_CASES.forEach(({ option, value }) => {
+        it(option, () => {
+          openOrdersTable({ limit: 1 });
+          cy.wait("@dataset");
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/visualizations-tabular/drillthroughs/column_extract_drill.cy.spec.js:71]:
```suggestion
```

File: e2e/test/scenarios/visualizations-tabular/drillthroughs/column_extract_drill.cy.spec.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+      extractColumnAndCheck({
+        column: "Min of Created At: Default",
+        option: "Year",
+        value: "2,022",
+      });
+    });
+
+    it("should handle duplicate expression names", () => {
+      openOrdersTable({ limit: 1 });
+      cy.wait("@dataset");
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/visualizations-tabular/drillthroughs/column_extract_drill.cy.spec.js:101]:
```suggestion
```

File: e2e/test/scenarios/visualizations-tabular/drillthroughs/column_extract_drill.cy.spec.js [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+      extractColumnAndCheck({
+        column: "Created At",
+        option: "Hour of day",
+        newColumn: "Hour of day_2",
+      });
+    });
+
+    it("should be able to modify the expression in the notebook editor", () => {
+      openOrdersTable({ limit: 1 });
+      cy.wait("@dataset");
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/visualizations-tabular/drillthroughs/column_extract_drill.cy.spec.js:116]:
```suggestion
```

File: e2e/test/scenarios/visualizations-tabular/drillthroughs/column_extract_drill.cy.spec.js [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+      cy.findByRole("gridcell", { name: "2,027" }).should("be.visible");
+    });
+
+    it("should use current user locale for string expressions", () => {
+      cy.request("GET", "/api/user/current").then(({ body: user }) => {
+        cy.request("PUT", `/api/user/${user.id}`, { locale: "de" });
+      });
+      openOrdersTable({ limit: 1 });
+      cy.wait("@dataset");
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/visualizations-tabular/drillthroughs/column_extract_drill.cy.spec.js:137]:
```suggestion
```

File: e2e/test/scenarios/visualizations-tabular/drillthroughs/column_extract_drill.cy.spec.js [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+        option: "Year",
+        value: "2,025",
+      });
+      cy.wait("@dataset");
+      openNotebook();
+      getNotebookStep("expression").findByText("Year").click();
+      enterCustomColumnDetails({ formula: "+ 2" });
+      popover().button("Update").click();
+      visualize();
+      cy.wait("@dataset");
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/visualizations-tabular/drillthroughs/column_extract_drill.cy.spec.js:128]:
```suggestion
```

**Review by nemanjaglumac — COMMENTED** [review]
`openOrdersTable` and `visualize` helpers are already waiting for the `@dataset`. There's no need to do it additionally in the test.

File: e2e/test/scenarios/visualizations-tabular/drillthroughs/column_extract_drill.cy.spec.js [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+      cy.findByRole("gridcell", { name: "2,027" }).should("be.visible");
+    });
+
+    it("should use current user locale for string expressions", () => {
```
Comment (kamilmielnik) [inline e2e/test/scenarios/visualizations-tabular/drillthroughs/column_extract_drill.cy.spec.js:132]: I think we should also check "Quarter" translations, it's something that could easily get overlooked.
For example in Polish and Russian quarters should be formatted as Roman numerals.

File: e2e/test/scenarios/visualizations-tabular/drillthroughs/column_extract_drill.cy.spec.js [hunk 7]
```diff
@@ -0,0 +1,157 @@
+import { SAMPLE_DATABASE } from "e2e/support/cypress_sample_database";
```
Comment (kamilmielnik) [inline e2e/test/scenarios/visualizations-tabular/drillthroughs/column_extract_drill.cy.spec.js:1]: It'd be great if new e2e tests were implemented in TS.

**Review by kamilmielnik — COMMENTED** [review]
> Different viz settings for each option - that's not the case now and I'm not sure if everything from the testing plan is correct about this

Let's discuss this in [Slack](https://metaboat.slack.com/archives/C0645JP1W81/p1709882492209779)

File: e2e/test/scenarios/visualizations-tabular/drillthroughs/column_extract_drill.cy.spec.js (hunk 6 above)
Comment (ranquild) [reply e2e/test/scenarios/visualizations-tabular/drillthroughs/column_extract_drill.cy.spec.js:127]: Apparently quarter names are always not localized neither on the FE or BE `(t/format "'Q'Q" t)`

**Comment by ranquild** [comment]
@nemanjaglumac for the record, this is not how "intercept" in cypress works. The problem arises when you have multiple intercepts for the same url:
```
cy.intercept("POST", "/api/dataset").as("dataset");
openOrdersTable(); // calls intercept for the same URL but different alias
// ... some action
cy.wait("@dataset"); // <-- this call would wait for the /api/dataset request made by `openOrdersTable` and will return immediately
```

We can solve this issue by using `intercept` closer to the action, and using unique aliases.

**Review by kamilmielnik — APPROVED** [review]

**Pull request merged**